scripts/check_compliance.py

```diff
@@ -668,6 +668,51 @@ def run(self):
             self.add_failure(msg.decode("utf-8"))
 
 
+class PyLint(ComplianceTest):
+    """
+    Runs pylint on all .py files, with a limited set of checks enabled. The
+    configuration is in the pylintrc file.
+    """
+    _name = "pylint"
+    _doc = "https://www.pylint.org/"
+
+    def run(self):
+        self.prepare(ZEPHYR_BASE)
+
+        # Path to pylint configuration file
+        pylintrc = os.path.join(os.path.dirname(__file__), "pylintrc")
+
+        # List of .py files added/modified by the commit(s).
+        #
+        # Skip create_board_img.py to work around a crash in pylint 2.2.2:
+        # https://github.com/PyCQA/pylint/issues/2906
+        py_files = git(
+            "diff", "--name-only", self.commit_range, "--", "*.py",
+            ":!boards/xtensa/intel_s1000_crb/support/create_board_img.py") \
+            .splitlines()
+
+        if not py_files:
+            return
+
+        try:
+            # Run pylint on added/modified .py files
+            process = subprocess.Popen(
+                ["pylint", "--rcfile=" + pylintrc] + py_files,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                cwd=ZEPHYR_BASE)
+        except FileNotFoundError:
+            self.error("pylint not found. Check that the directory it's in is "
+                       "listed in the PATH environment variable.")
+        except OSError as e:
+            self.error("Failed to run pylint: " + str(e))
+
+        stdout, stderr = process.communicate()
```

Comment:
The more recent (Python 3.5) `subprocess.run()` saves you the `.communicate()` and manual utf-8 decoding.

Reply:
The `encoding` parameter to `subprocess.run()` was added in Python 3.6, and that's too recent, so won't get rid of the decoding I think. Think we might still support Python 3.4 too.

I like the high-level interface personally. Once you've gotten it right, it's easy to tweak.

Reply:
> Think we might still support Python 3.4 too.

Correct: https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/python.cmake#L16

I'd like to drop 3.4 support, as it's EOL in upstream Python since March [*], and I've seen a fair bit of useful stuff added in 3.5 that requires workarounds or less elegant methods in 3.4. I haven't been able to figure out if we've got any older RHEL or Ubuntu 14.04 users out there that would complain, though.

[*] https://www.python.org/downloads/release/python-3410/